F/finufft/build_tarballs.jl

```diff
@@ -19,11 +21,16 @@ make lib CFLAGS="-fopenmp -fPIC -O3 -funroll-loops -fcx-limited-range -Iinclude"
 mv lib/libfinufft.so "${libdir}/libfinufft.${dlext}"
 """
 
-# These are the platforms we will build for by default, unless further
-# platforms are passed in on the command line
-platforms = supported_platforms()
+# Build for all supported platforms
+platforms_x86_64 = filter(p -> p.tags["arch"]=="x86_64", supported_platforms())
+platforms_other  = filter(p -> p.tags["arch"]!="x86_64", supported_platforms())
 
+# Expand for microarchitectures on x86_64 (library doesn't have CPU dispatching)
+# Tests on Linux/x86_64 yielded a slow binary with avx512 for some reason, so disable that
+platforms_x86_64 = expand_microarchitectures(platforms_x86_64)
+platforms_x86_64 = filter(p -> p.tags["march"] != "avx512", platforms_x86_64)
 
+platforms = vcat(platforms_x86_64, platforms_other)
```

Comment:
We should probably add a way to automatically expand microarchitectures only for specified architectures, it isn't uncommon that the libraries do this kind of optimisation only for x86_64.

If you were worried about the failures on aarch64, those are bugs that we have to fix anyway (I already have some fixes locally but have to test something else)

Reply:
Well yes, I was trying to circumvent the failures :)

But in any case, the goal of the microarchitecture expansion is to get binaries with AVX support, so limiting it to x86_64 makes a lot of sense in any case.

Reply:
For the record, I'm slowly making some progress towards #4252.  https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/205 fixed the aarch64 flags (not relevant here, but important before moving microarchitectures forward) and with https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/232 it is now possible to restrict the expansion with an additional argument:
```julia
julia> platforms_x86_64 = filter(p -> p.tags["arch"]=="x86_64", supported_platforms());
       platforms_other  = filter(p -> p.tags["arch"]!="x86_64", supported_platforms());
       platforms_x86_64 = expand_microarchitectures(platforms_x86_64);
       platforms_x86_64 = filter(p -> p.tags["march"] != "avx512", platforms_x86_64);
       platforms = vcat(platforms_x86_64, platforms_other)

26-element Vector{Platform}:
 Linux x86_64 {libc=glibc, march=x86_64}
 Linux x86_64 {libc=glibc, march=avx}
 Linux x86_64 {libc=glibc, march=avx2}
 Linux x86_64 {libc=musl, march=x86_64}
 Linux x86_64 {libc=musl, march=avx}
 Linux x86_64 {libc=musl, march=avx2}
 macOS x86_64 {march=x86_64}
 macOS x86_64 {march=avx}
 macOS x86_64 {march=avx2}
 FreeBSD x86_64 {march=x86_64}
 FreeBSD x86_64 {march=avx}
 FreeBSD x86_64 {march=avx2}
 Windows x86_64 {march=x86_64}
 Windows x86_64 {march=avx}
 Windows x86_64 {march=avx2}
 Linux i686 {libc=glibc}
 Linux aarch64 {libc=glibc}
 Linux armv6l {call_abi=eabihf, libc=glibc}
 Linux armv7l {call_abi=eabihf, libc=glibc}
 Linux powerpc64le {libc=glibc}
 Linux i686 {libc=musl}
 Linux aarch64 {libc=musl}
 Linux armv6l {call_abi=eabihf, libc=musl}
 Linux armv7l {call_abi=eabihf, libc=musl}
 macOS aarch64
 Windows i686

julia> expand_microarchitectures(supported_platforms(), ["x86_64", "avx", "avx2"])
26-element Vector{Platform}:
 Linux i686 {libc=glibc}
 Linux x86_64 {libc=glibc, march=x86_64}
 Linux x86_64 {libc=glibc, march=avx}
 Linux x86_64 {libc=glibc, march=avx2}
 Linux aarch64 {libc=glibc}
 Linux armv6l {call_abi=eabihf, libc=glibc}
 Linux armv7l {call_abi=eabihf, libc=glibc}
 Linux powerpc64le {libc=glibc}
 Linux i686 {libc=musl}
 Linux x86_64 {libc=musl, march=x86_64}
 Linux x86_64 {libc=musl, march=avx}
 Linux x86_64 {libc=musl, march=avx2}
 Linux aarch64 {libc=musl}
 Linux armv6l {call_abi=eabihf, libc=musl}
 Linux armv7l {call_abi=eabihf, libc=musl}
 macOS x86_64 {march=x86_64}
 macOS x86_64 {march=avx}
 macOS x86_64 {march=avx2}
 macOS aarch64
 FreeBSD x86_64 {march=x86_64}
 FreeBSD x86_64 {march=avx}
 FreeBSD x86_64 {march=avx2}
 Windows i686
 Windows x86_64 {march=x86_64}
 Windows x86_64 {march=avx}
 Windows x86_64 {march=avx2}

julia> all(in(platforms), expand_microarchitectures(supported_platforms(), ["x86_64", "avx", "avx2"]))
true
```